client/components/data/query-purchase-cancellation-offers/index.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCancellationOffers } from 'calypso/state/cancellation-offers/actions';
+import isFetchingCancellationOffers from 'calypso/state/cancellation-offers/selectors/is-fetching-cancellation-offers';
+
+interface OwnProps {
+	siteId?: number;
+	purchaseId?: number;
```

Comment:
In the README we're mentioning that both props are required, however here we mark them optional. I understand the reasoning that here we would like to handle `nulls` inside the component - however, I would say the clearer approach would be to ensure these values are defined before starting to render the component.

Reply:
Optionally, we can consider including `isLoading` / `isEnabled` property from the top of the component. I understand this approach to be very similar to react-query, and this is how they handle it there:

https://tanstack.com/query/v4/docs/guides/dependent-queries